App/src/pages/RegularGroups.tsx

```diff
@@ -0,0 +1,83 @@
+import { Container, Table, Thead, Tbody, Tr, Th, Td } from "@chakra-ui/react";
+import { useContext, useEffect, useState } from "react";
+import { GetAllGroups, GroupDTO, MultiGroupResponseDTO } from "../services/group";
+import { UserContext } from "../App";
+import { UserDTO } from "../services/user";
+import Header from "../shared/Header";
+
+export default function DisplayGroups() {
+    const [myGroups, setMyGroups] = useState<GroupDTO[]>([]);
+    const user = useContext(UserContext);
+    const userDTO: UserDTO = { firstName: user.firstName, id: "7", lastName: user.lastName, email: user.email };
+    useEffect(() => {
+        AllGroups();
+        console.log(user);
+    }, []);
+
+    function AllGroups() {
+        GetAllGroups()
+            .then(
+                (res: MultiGroupResponseDTO) => {
+                    setMyGroups(res.data);
+                    setMyGroups(res.data.filter(membership));
+                },
+                (rej) => {
+                    console.log(rej);
+                }
+            )
+            .catch((err) => {
+                console.log(err);
+            });
+    }
+    function membership(GroupDTO: GroupDTO): boolean {
+        if (GroupDTO.members.includes(userDTO)) {
```

Comment:
1: You can just make this method a 1-liner with `return GroupDTO.members.includes(userDTO);`, or not even have this method and just write it as an anonymous function in your setMyGroups, since you only call it once.

2: This call will ALWAYS return false, because you're comparing objects which exist in different locations in memory, and these will always be said to be different objects. I think this might work better:
```
return GroupDTO.members.map(member => member.id).includes(user.id);
```
Essentially what we're doing is we extract an array of member IDs with the map function then we see if the new array contains the id of the currently logged in user.

Reply:
ok, fixed